src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -2856,8 +2855,7 @@ void InitSolarHeatGains(EnergyPlusData &state)
             int const firstSurfOpaq = state.dataHeatBal->Zone(zoneNum).OpaqOrIntMassSurfaceFirst;
             int const lastSurfOpaq = state.dataHeatBal->Zone(zoneNum).OpaqOrIntMassSurfaceLast;
             for (int SurfNum = firstSurfOpaq; SurfNum <= lastSurfOpaq; ++SurfNum) {
-                int ConstrNum = Surface(SurfNum).Construction;
-                if (state.dataSurface->SurfWinStormWinFlag(SurfNum) == 1) ConstrNum = Surface(SurfNum).StormWinConstruction;
+                int ConstrNum = state.dataSurface->Surface(SurfNum).Construction;
```

Comment:
Originally I intended to only use `SurfWinActiveConstruction` to overwrite the exterior window constr index. The opaq surface loop would not need to be overwritten. It may be possible to use `SurfActiveConstruction` to overwrite any use case of `Surface(i).Construction`, because this logic 
```
if (state.dataSurface->SurfWinStormWinFlag(SurfNum) == 1) {
    state.dataSurface->SurfWinActiveConstruction(SurfNum) = state.dataSurface->SurfWinStormWinConstr(SurfNum);
} else {
    state.dataSurface->SurfWinActiveConstruction(SurfNum) = state.dataSurface->Surface(SurfNum).Construction;
}
```
is set at the beginning of the simulation loop.

However, I don't know if the storm window index, when exists, is supposed to overwrite any construction index in any calculations. For example, I noticed that only when `WindowModelType(SurfNum) == Window5DetailedModel`, the constr index is overwritten, but for BSDF and EQL windows, the construction are still set as the original one (`Surface(SurfNum).Construction`). I checked the IO Ref and IDD notes and I don't see any restriction to attach a storm window when `ComplexFenestration` or `EQL layers` are applied.

Reply:
But this is inside a loop for opaque surfaces, so state.dataSurface->SurfWinStormWinFlag(SurfNum) does not apply here. I think line 2860 ` if (state.dataSurface->SurfWinStormWinFlag(SurfNum) == 1) ConstrNum = Surface(SurfNum).StormWinConstruction;` can be deleted.

Reply:
@mjwitte Regarding this, I just mocked up a test case to apply StormWindow to an exterior window (idf file attached) and no error checkings prevent me from doing so. The storm index is assigned to the window, but triggers a segfault [here](https://github.com/NREL/EnergyPlus/blob/6334396b7bcebeb5eb801747eff3d316911f0956/src/EnergyPlus/HeatBalanceSurfaceManager.cc#L7152) to get the `Construct(ConstrNum).EQLConsPtr` for the storm window construction. So I guess I should fix it by checking the window model [here](https://github.com/NREL/EnergyPlus/blob/6334396b7bcebeb5eb801747eff3d316911f0956/src/EnergyPlus/HeatBalanceSurfaceManager.cc#L7133) and wherever the storm window is overwriting the original construction number. This should be an existing defect in the develop branch.

[EquivalentLayerWindow_test.idf.txt](https://github.com/NREL/EnergyPlus/files/6281192/EquivalentLayerWindow_test.idf.txt)